ts/src/test/test.ts

```diff
@@ -352,33 +359,24 @@ export default class testMainClass extends baseMainTestClass {
         // promises.push (testThrottle ());
         const results = await Promise.all (promises);
         // now count which test-methods retuned `false` from "testSafe" and dump that info below
-        const errors = [];
-        for (let i = 0; i < testNames.length; i++) {
-            if (!results[i]) {
-                errors.push (testNames[i]);
-            }
-        }
         if (this.info) {
+            const errors = [];
+            for (let i = 0; i < testNames.length; i++) {
+                if (!results[i]) {
+                    errors.push (testNames[i]);
+                }
+            }
             // we don't throw exception for public-tests, see comments under 'testSafe' method
             let failedMsg = '';
             if (errors.length) {
                 failedMsg = ' | Failed methods: ' + errors.join (', ');
             }
-            dump (this.addPadding ('[INFO:PUBLIC_TESTS_DONE]' + market['type'] + failedMsg, 25), exchange.id);
+            dump (this.addPadding ('[INFO:PUBLIC_TESTS_END] ' + market['type'] + failedMsg, 25), exchange.id);
         }
     }
 
     async loadExchange (exchange) {
-        try {
-            await this.testSafe ('loadMarkets', exchange, [], true);
-        } catch (e) {
-            if (e instanceof OnMaintenance) {
-                dump ('[SKIPPED] Exchange is on maintenance', exchange.id);
-                exitScript ();
-            }
-            // if excepion is not maintenance (and therefore, neither temporary connection exceptiions, defined in `testSafe`) then throw exception as is, and the caller method will handle that
-            throw e;
-        }
+        await this.testSafe ('loadMarkets', exchange, [], true);
```

Comment:
no longer need to be inside try-catch, as the logic was moved directly in `testSafe`.